### PR TITLE
allow same partner on both sides of a relation

### PR DESCRIPTION
**Pull request by charbeljc**
- add boolean field `ResPartnerRelationType.allow_self`
- add related field `ResPartnerRelation.allow_self`

### CODE REVIEW

File: partner_relations/model/res_partner_relation_type.py [hunk 1]
```diff
@@ -57,6 +57,9 @@ class ResPartnerRelationType(models.Model):
         'res.partner.category',
         'Right partner category',
     )
+    allow_self = fields.Boolean(
+        'Allow both sides to be the same'
```
Comment (pedrobaeza) [inline partner_relations/model/res_partner_relation_type.py]: Put `default=False`

**Comment by pedrobaeza** [comment]
:+1:

**Comment by gurneyalex** [comment]
@charbeljc can you add an automated test for this new feature?

**Comment by antespi** [comment]
:+1:

**Comment by hbrunn** [comment]
:+1:

**Pull request merged**